src/eva/vision/losses/dice.py

```diff
@@ -12,29 +14,94 @@ class DiceLoss(losses.DiceLoss):  # type: ignore
     Extends the implementation from MONAI
         - to support semantic target labels (meaning targets of shape BHW)
         - to support `ignore_index` functionality
+        - accept weight argument in list format
     """
 
-    def __init__(self, *args, ignore_index: int | None = None, **kwargs) -> None:
-        """Initialize the DiceLoss with support for ignore_index.
+    def __init__(
+        self,
+        *args,
+        ignore_index: int | None = None,
+        weight: Sequence[float] | torch.Tensor | None = None,
+        **kwargs,
+    ) -> None:
+        """Initialize the DiceLoss.
 
         Args:
             args: Positional arguments from the base class.
             ignore_index: Specifies a target value that is ignored and
                 does not contribute to the input gradient.
+            weight: A list of weights to assign to each class.
             kwargs: Key-word arguments from the base class.
         """
-        super().__init__(*args, **kwargs)
+        if weight is not None and not isinstance(weight, torch.Tensor):
+            weight = torch.tensor(weight)
+
+        super().__init__(*args, **kwargs, weight=weight)
 
         self.ignore_index = ignore_index
 
     @override
     def forward(self, inputs: torch.Tensor, targets: torch.Tensor) -> torch.Tensor:  # noqa
-        if self.ignore_index is not None:
-            mask = targets != self.ignore_index
-            targets = targets * mask
-            inputs = torch.mul(inputs, mask.unsqueeze(1) if mask.ndim == 3 else mask)
+        inputs, targets = _apply_ignore_index(inputs, targets, self.ignore_index)
+        targets = _to_one_hot(targets, num_classes=inputs.shape[1])
 
         if targets.ndim == 3:
             targets = one_hot(targets[:, None, ...], num_classes=inputs.shape[1])
 
         return super().forward(inputs, targets)
+
+
+class DiceCELoss(losses.dice.DiceCELoss):
```

Comment:
no strong feelings but usually prefer spelling names out to make it more clear if looked at without context. Here could be `DiceCrossEntropyLoss`

Reply:
Good point. In this instance I kept the original name, to make it more clear that those are just wrappers around commonly used loss functions from other libraries (monai in this case).